cmd/createorupdate/update.go

```diff
@@ -11,7 +12,19 @@ import (
 	"github.com/openshift/openshift-azure/pkg/log"
 )
 
-func update(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.Plugin) error {
+func update(ctx context.Context, cs, oldCs *api.OpenShiftManagedCluster, p api.Plugin, azuredeploy []byte) error {
+	scaleOps := isScaleOp(cs, oldCs)
+	for _, scale := range scaleOps {
+		if scale.down {
+			// TODO: If we are scaling down, we need to drain first.
+		}
+	}
+
```

Comment:
Here, loop roles and for each, record which VMs currently exist pre-deploy

Reply:
done